### ISSUE TITLE
Error on training (pxEval_maximizeFMeasure, ValueError: max() arg is an empty sequence)

### ISSUE THREAD

**Issue by cxfire16** [MarvinTeichmann/KittiSeg#153]
My goal: train with a different dataset (just 1 class same as the road dataset)
Issue: I encounter error during the training (detailed below). I tried tracing the error, but to no avail I can't seem to see the problem. Also I seem to notice that Acc. values are near perfect (see logs below) leading me to think it's the data.
dataset: custom and currently in progress. currently have 21 images. I wish also to know if this could be the cause of the error; that there's too tiny dataset. I divided the 21 images into 17 for training and 4 for validation.

**Configuration**
Used default, followed the easy step. prepared the dataset and trained using KittiSeg.json, also tried KittiSeg_VGG.json, same error.

**The error**

2018-01-24 16:36:27.104698: I tensorflow/core/kernels/logging_ops.cc:79] Shape of dropout_1/mul:0[1 22 22 4096]
2018-01-24 16:36:27.106815: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore2[1 44 44 2]
2018-01-24 16:36:27.107433: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore4[1 88 88 2]
2018-01-24 16:36:27.140504: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore32[1 700 700 2]
2018-01-24 16:36:28,778 INFO Step 0/12000: loss = 0.48; lr = 1.00e-05; 0.061 sec (per Batch); 16.4 imgs/sec
2018-01-24 16:36:28,968 INFO       (raw)  Acc. : 0.52, xentropy: 0.16, weight_loss: 0.38
2018-01-24 16:36:28,968 INFO    (smooth)  Acc. : 0.52, xentropy: 0.16, weight_loss: 0.38
2018-01-24 16:36:42,336 INFO Step 20/12000: loss = 0.38; lr = 1.00e-05; 0.267 sec (per Batch); 3.7 imgs/sec
2018-01-24 16:36:42,481 INFO       (raw)  Acc. : 1.00, xentropy: 0.00, weight_loss: 0.38
2018-01-24 16:36:42,482 INFO    (smooth)  Acc. : 0.54, xentropy: 0.15, weight_loss: 0.38
2018-01-24 16:36:51,777 INFO Step 40/12000: loss = 0.38; lr = 1.00e-05; 0.186 sec (per Batch); 5.4 imgs/sec
2018-01-24 16:36:51,923 INFO       (raw)  Acc. : 0.99, xentropy: 0.00, weight_loss: 0.38
2018-01-24 16:36:51,923 INFO    (smooth)  Acc. : 0.57, xentropy: 0.14, weight_loss: 0.38
2018-01-24 16:37:01,229 INFO Step 60/12000: loss = 0.38; lr = 1.00e-05; 0.186 sec (per Batch); 5.4 imgs/sec
2018-01-24 16:37:01,375 INFO       (raw)  Acc. : 1.00, xentropy: 0.00, weight_loss: 0.38
2018-01-24 16:37:01,375 INFO    (smooth)  Acc. : 0.59, xentropy: 0.14, weight_loss: 0.38
2018-01-24 16:37:10,676 INFO Step 80/12000: loss = 0.38; lr = 1.00e-05; 0.186 sec (per Batch); 5.4 imgs/sec
2018-01-24 16:37:10,824 INFO       (raw)  Acc. : 1.00, xentropy: 0.00, weight_loss: 0.38
2018-01-24 16:37:10,824 INFO    (smooth)  Acc. : 0.61, xentropy: 0.13, weight_loss: 0.38
2018-01-24 16:37:20,122 INFO Step 100/12000: loss = 0.38; lr = 1.00e-05; 0.186 sec (per Batch); 5.4 imgs/sec
2018-01-24 16:37:20,270 INFO       (raw)  Acc. : 1.00, xentropy: 0.00, weight_loss: 0.38
2018-01-24 16:37:20,271 INFO    (smooth)  Acc. : 0.63, xentropy: 0.12, weight_loss: 0.38
2018-01-24 16:37:23,982 INFO Running Evaluation Script.
2018-01-24 16:37:24.178556: I tensorflow/core/kernels/logging_ops.cc:79] Shape of Validation/Validation/fc7/Relu:0[1 22 22 4096]
2018-01-24 16:37:24.180357: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore2[1 44 44 2]
2018-01-24 16:37:24.181009: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore4[1 88 88 2]
2018-01-24 16:37:24.182286: I tensorflow/core/kernels/logging_ops.cc:79] Shape of upscore32[1 700 700 2]
/home/chuck/chuck/venvs/tensorgpu-1/data/testWallDetect/incl/seg_utils/seg_utils.py:119: RuntimeWarning: invalid value encountered in true_divide
  recall = totalTP / float( totalPosNum )
/home/chuck/chuck/venvs/tensorgpu-1/data/testWallDetect/incl/seg_utils/seg_utils.py:135: RuntimeWarning: invalid value encountered in greater_equal
  ind = np.where(recall>=i)
2018-01-24 16:37:26.961810: W tensorflow/core/kernels/queue_base.cc:295] _0_Queues/fifo_queue: Skipping cancelled enqueue attempt with queue not closed
Traceback (most recent call last):
  File "train.py", line 131, in <module>
    tf.app.run()
  File "/home/chuck/chuck/venvs/tensorgpu-1/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "train.py", line 127, in main
    train.do_training(hypes)
  File "incl/tensorvision/train.py", line 395, in do_training
    run_training(hypes, modules, tv_graph, tv_sess)
  File "incl/tensorvision/train.py", line 288, in run_training
    hypes, sess, tv_graph['image_pl'], tv_graph['inf_out'])
  File "/home/chuck/chuck/venvs/tensorgpu-1/data/testWallDetect/hypes/../evals/kitti_eval.py", line 135, in evaluate
    total_posnum, total_negnum, total_fn, total_fp, thresh=thresh)
  File "/home/chuck/chuck/venvs/tensorgpu-1/data/testWallDetect/incl/seg_utils/seg_utils.py", line 138, in pxEval_maximizeFMeasure
    pmax = max(precision[ind])
ValueError: max() arg is an empty sequence

**Comment by cxfire16** [comment]
I'm closing this as duplicate of #110

**Comment by Ratnadeep22** [comment]
Has anyone solved this error?

**Comment by pouyashiri** [comment]
This error means that both recall and precision are 0 for all images. In my case, it had something to do with the color specified for background and road classes. Note that the images are read using RGB mode (scipy.misc.imread). Also, I had anotated my data using labelme and expected the red color to be [0,0,255], but I checked the images and the red pixels were [0,0,128] instead.
I hope this helps.
